### ISSUE TITLE
RLC unsoundness at (pseudo?) assignments to `@MustCallUnknown`

### ISSUE THREAD

**Issue by Calvin-L** [typetools/checker-framework#6990]
The RLC reports no violations on this code, although the `resource` parameter is clearly dropped without being closed:
```java
import org.checkerframework.checker.mustcall.qual.MustCallUnknown;
import org.checkerframework.checker.mustcall.qual.Owning;

import java.io.Closeable;

public class DropOwning {

  public void f(@Owning Closeable resource) {
    drop(resource);
  }

  private void drop(@Owning @MustCallUnknown Object resource) {
  }

}
```
The manual [suggests](https://checkerframework.org/manual/#resource-leak-generic-unknown) that this code should result in a `required.method.not.known` warning, although that doesn't seem to be the case.

**Comment by kelloggm** [comment]
This is definitely an unsoundness bug. Thanks for the report, @Calvin-L.

> The manual suggests that this code should result in a required.method.not.known warning, although that doesn't seem to be the case.

I agree that that's the right behavior here. An `@Owning` pointer with that type should always produce that warning, so I'm not sure what's going on here. I'll investigate.
